atc/scheduler/scheduler.go

```diff
@@ -93,9 +93,34 @@ func (s *Scheduler) ensurePendingBuildExists(
 		}
 	}
 
+	jobNeedsManualTrigger := firstOccurrenceInInputMapping(inputMapping) && jobIsManuallyTriggered(job)
+	if job.NeedsManualTrigger() != jobNeedsManualTrigger {
+		if err := job.SetNeedsManualTrigger(jobNeedsManualTrigger); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func firstOccurrenceInInputMapping(inputMapping algorithm.InputMapping) bool {
+	for _, iM := range inputMapping {
+		if iM.FirstOccurrence {
+			return true
+		}
+	}
+	return false
+}
+
+func jobIsManuallyTriggered(job db.Job) bool {
```

Comment:
I think might still be useful to mark something as 'needs triggering' even if it has some inputs with `trigger: true`. 🤔

Reply:
In what cases?

If we were to do that the frontend code would be more complicated, as we would have to again export the last version of which a input was running with to determine if a UI change would be necessary.

Reply:
Sometimes I configure jobs to only trigger on the inputs that are closer to my project, and leave things like non-critical external dependencies as `trigger: false`. For example, the four jobs in the second column in [our pipeline](https://ci.concourse-ci.org/teams/main/pipelines/concourse) have a non-trigger `postgres-image` input.

I think it would still be useful to surface to the user when there are new inputs available for that resource, even though the job can technically trigger on its own and pick it up. It may be something I notice shortly before shipping, and I may decide to pick up the new dependency for example.

I see this as an indicator of "if i were to trigger the job, would it have new inputs?" (#269) - whereas right now it seems to be more specifically "this job never triggers on its own, so let me know when it has new inputs to trigger". Which is actually the first time Concourse would special-case "manual-trigger-only" jobs.

Reply:
Ok, that seems like a nice use case.

I guess a concern would be that the UI might "flash" if doing this?

I.e 

1. Schedule iterates over jobs, one of which have one `trigger: true` and one `trigger: false` input, where the both inputs have first occurring versions.
2. The job gets updated with a `new-non-used-input` bool.
3. The ui fetches jobs data, and renderers some indicator
4. Time goes by
5. The job starts executing due to the `trigger: true` job
6. The indicator is removed since the job is now running.

So if looking at the pipeline dashboard view, you would see a job flash between states.